Sources/RegexBuilder/Variadics.swift

```diff
@@ -4172,6 +4172,421 @@ extension TryCapture {
   }
 }
 
+@available(SwiftStdlib 5.7, *)
+extension RegexComponent {
```

Comment:
Do we want this on every `RegexComponent`? What about an internal-ish protocol for constructions that add a layer of optionality, so that these don't appear on `String`, `Regex`, etc?